### DISCUSSION TITLE
Getting dictionary of all data in a database reference

### DISCUSSION THREAD

**Discussion by macktns** [GodotNuts/GodotFirebase#147] [Q&A] [answered]
Hello! Having trouble with this as the chat example seems to be listening for new data rather than retrieving all data. I Have successfully pushed data to the database. I assumed I would use get_data on the same database reference to get a dictionary of the data but I get an error "Invalid get index '_root' (on base: 'Dictionary')." What am I doing wrong or what could be wrong? 
pastebin of my extremely minimal project https://pastebin.com/RS83p9Hw

**Comment by WolfgangSenff** [comment, marked as answer]
Can't say with certainty, but my immediate thought is that you haven't actually waited for it to pull data back. You should only be calling things after the reference has completed initializing. More importantly, the example you're looking at is for the part of the chat app which *pushes* the data, not the one which listens for it. (It will listen for it, but in this case it is ignored.) What you're looking for regarding what to do is: https://github.com/GodotNuts/FirebaseDemo-Chat/blob/main/Chat.gd#L15 - No need to do a get_data if you're reading the data in the new_data_update signal which returns you the data. This one will return the entire set of data at the very beginning for you as well, but from that point onward, only updates.

**Reply by macktns** [reply]
Firstly this project is amazing and I really appreciate you guys. I want to use it so badly and I'm so close! Ok I tried to use new_data_update. Here is the very short script. This project is nothing but the addon and this script. https://pastebin.com/gJFa0gtP  . The error I get with this script is "E 0:00:02.460   _resolve_hostname: getaddrinfo failed! Cannot resolve hostname.
  <C++ Source>  drivers/unix/ip_unix.cpp:112 @ _resolve_hostname()".
I am still able to push data without a problem. I have all the server info in both the script and the project settings. 
I also tried adding a timer and waiting 10 seconds before requesting the data but no difference
Running on windows 10. No sort of firewalls are running besides built in windows

**Reply by macktns** [reply]
Ok I did two things and now it is working. I upgraded to a blaze plan first. This did not work at least within the first 20 minutes or so. A few hours later I made a new project on a different computer setup in the exact same way and it suddenly works.
